src/logic.service.ts

```diff
@@ -33,11 +34,13 @@ export const handleTransaction = async (transaction: unknown): Promise<void> =>
     default:
       break;
   }
+  apmTransaction?.end();
 };
 
 const handlePain001 = async (transaction: Pain001): Promise<void> => {
   loggerService.log('Start - Handle transaction data');
-  const span = apm.startSpan('Handle transaction data');
+  const span = apm.startSpan('transaction.pain001');
```

Comment:
Where do you close this span?

Reply:
On line [#117](https://github.com/rtkay123/data-preparation/blob/9606f9f8f5de65ed7d53f3c91f3f0105f4ffc5cd/src/logic.service.ts#L117)